README.md

```diff
@@ -1,339 +1,196 @@
+```tsx
+import { useLiveQuery } from '@tanstack/react-optimistic'
 
-Preloads data for a collection before rendering components.
+const Todos = () => {
+  const { data: todos } = useLiveQuery(query =>
+    // You can query across collections with joins, aggregates, etc.
+    query
+      .from({ todoCollection })
+      .where('@completed', '=', false)
+      .select('@id', '@text')
+      .keyBy('@id')
+  )
 
-```typescript
-await preloadCollection({
-  id: 'todos',
-  sync: { /* sync configuration */ },
-  mutationFn: { /* mutation functions */ },
-  schema: /* optional schema */
-});
+  return <List items={todos} />
+}
 ```
 
-Features:
+Make writes using transactional mutations:
 
-1. Returns a promise that resolves when the first sync commit is complete
-2. Shares the same collection instance with `useCollection`
-3. Handles already-loaded collections by returning immediately
-4. Avoids duplicate initialization when called multiple times with the same ID
+```ts
+import type { Collection } from '@tanstack/optimistic'
+import type { MutationFn, PendingMutation } from '@tanstack/react-optimistic'
+import { createTransaction } from '@tanstack/react-optimistic'
 
-### Data Operations
+const filterOutCollection = (mutation: PendingMutation) => {
+  const { collection: _, ...rest } = mutation
 
-#### Insert
-
-```typescript
-// Insert a single item
-insert({ text: "Buy groceries", completed: false })
-
-// Insert multiple items
-insert([
-  { text: "Buy groceries", completed: false },
-  { text: "Walk dog", completed: false },
-])
+  return rest
+}
 
-// Insert with custom key
-insert({ text: "Buy groceries" }, { key: "grocery-task" })
-```
+// Transactions are created with a mutationFn that handles their mutations.
+// In this case, we POST them to the server.
+const tx = createTransaction({
+  mutationFn: async ({ transaction }) => {
+    const payload = transaction.mutations.map(filterOutCollection)
+    const response = await fetch('https://example.com/your-api', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload)
+    })
 
-#### Update
+    if (!response.ok) {
+      // Throwing an error will rollback the optimistic state.
+      throw new Error(`HTTP Error: ${response.status}`)
+    }
 
-We use a proxy to capture updates as immutable draft optimistic updates.
+    const result = await response.json()
 
-```typescript
-// Update a single item
-update(todo, (draft) => {
-  draft.completed = true
+    // Wait for the transaction to be synced back from the server
+    // before discarding the optimistic state.
+    const collection: Collection = transaction.mutations[0]!.collection
+    await collection.config.sync.awaitTxid(result.txid)
+  }
 })
 
-// Update multiple items
-update([todo1, todo2], (drafts) => {
-  drafts.forEach((draft) => {
-    draft.completed = true
+// Triggers the mutationFn
+tx.mutate(() =>
+  // Applies the local optimistic state
```

Comment:
I'm eyeballing these comments wondering if they're wrong. I.e.: when does the local optimistic state get applied?

Reply:
Yeah this is correct. They're applied immediately